Merge Restructure code to Development

ansariDataamanulla opened the pull request:
**Code Restructuring:**
The file and folder structure has been updated to follow an instance-based structure, making it easier to manage and build instances. This update improves organization and scalability, allowing for faster, more efficient builds on an instance-wise basis.
Updated and changed the scope of Google Drive backup integration to enhance storage efficiency and backup reliability.
**Unique Name Changes:**
Merged the Babuni implementation with the Bebbo codebase, making necessary adjustments to align with the new naming convention for taxonomyIds.
**Bug Fixes:**
Fixed font rendering issues across different devices for consistent typography.
Resolved minor bugs affecting the overall user experience, improving stability and performance.

Akhror commented on .gitignore:
```diff
@@ -44,22 +45,25 @@ ios/ParentBuddyApp.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks
-
-
-# sonar qube 
-.scannerwork
```
.scannerwork folder should be excluded from the repository. It contains temporary reports or logs which is not needed for building or running the project and can be regenerated during analysis.

ansariDataamanulla replied:
Okay

Akhror replied:
Let's please review the below refined gitignore and replace gitignore in the project.

# Node.js dependencies
node_modules/
npm-debug.log
yarn-error.log

# Build artifacts
*.lock
*.bundle
android/app/src/main/assets/index.android.bundle
android/app/build/
ios/build/
ios/tmp.xcconfig

# Android-specific files
android/.gradle/
android/.idea/
android/app/src/prod/res/values/strings.xml
android/gradle.properties
local.properties
google-services*.json

# iOS-specific files
ios/Pods/
ios/ParentBuddyApp.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist
ios/ParentBuddyApp.xcodeproj/xcuserdata/*.xcuserdatad/xcschemes/xcschememanagement.plist
ios/GoogleServices/**/GoogleService-Info*.plist
ios/**/Info.plist
ios/**-Info.plist


# Environment files
.env*

# BUCK build system
buck-out/
\.buckd/

# Keystore files
*.keystore
!debug.keystore

# Gradle & other properties files
.gradle/
*.properties

# Miscellaneous
.DS_Store
*.hprof
data.realm
data.realm.management/

ansariDataamanulla replied:
Okay i'll replace

Akhror commented on .gitignore:
```diff
@@ -14,6 +14,7 @@ buck-out/
 !debug.keystore
 # .env added
 .env
+.env.*
```
As we agreed we will add one env file for dev environment.

Akhror commented on .gitignore:
```diff
@@ -44,22 +45,25 @@ ios/ParentBuddyApp.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks
 ios/ParentBuddyApp.xcodeproj/xcuserdata/kalyanikaushik.xcuserdatad/xcschemes/xcschememanagement.plist
 ios/GoogleServices/Xk/GoogleService-Info_old.plist
 ios/GoogleServices/Production/GoogleService-Info_new.plist
+ios/GoogleServices/MerhabaBebek/GoogleService-Info.plist
```
Instead of ignoring plist files one by one we should ignore them by file extension:

`*.plist`

ansariDataamanulla replied:
no we cant ignore file extension info.plist file also there

Akhror replied:
We can exclude it like
*.plist
!**/Info.plist

Akhror commented on .gitignore:
```diff
@@ -44,22 +45,25 @@ ios/ParentBuddyApp.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks
 ios/ParentBuddyAppXk-Info.plist
+ios/ParentBuddyAppBangla-Info.plist
+ios/ParentBuddyAppBanglaDev-Info.plist
 ios/GoogleServices/Development/GoogleService-Info.plist
```
As per our naming convention these folders should be:
bebboDevelopment instead of Development
bebbo instead of Production
babuni instead of Bangla
babuniDevelopment instead of BanglaDevelopment
foleja instead of Xk
folejaDevelopment instead of XkDevelopment

Akhror commented on ExportOptionsBabuni.plist:
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Can we move
```
ExportOptions.plist
ExportOptionsBabuni.plist
tsconfig.babuni.json
tsconfig.bebbo.json
tsconfig.merhabaBebek.json 
```
files from root directory to their respective instance folders?

ansariDataamanulla replied:
NO we cant

Akhror commented on android/app/src/bangla/AndroidManifest.xml:
```diff
@@ -0,0 +1,53 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+  package="org.unicef.ecar.bebbo">
```
As we agreed in the naming convention instance specific files and folders should be named as:
babuni not bangla
babuniDev not banglastaging, not BanglaDevelopment
bebbo not prod, not Production
bebboDev not prodstaging, not Development
foleja not xk
folejaDev not xkstaging, not XkDevelopment.

Folders to look at:
parenting-app-bebbo-mobile\ios
parenting-app-bebbo-mobile\ios\GoogleServices
parenting-app-bebbo-mobile\ios\LottieSplash
parenting-app-bebbo-mobile\ios\ParentBuddyApp
parenting-app-bebbo-mobile\android\app
parenting-app-bebbo-mobile\android\app\src

ansariDataamanulla replied:
@Akhror do it later

Akhror commented on android/app/src/banglastaging/res/values-bn/strings.xml:
```diff
@@ -0,0 +1,3 @@
+<resources>
+    <string name="app_name">بيبو</string>
```
Why app name is Bebbo, should change it to Babuni (বাবুনি)

Akhror replied:
Marked as resolved but I do not see the change.

Akhror commented on android/app/src/merhababebek/AndroidManifest.xml:
```diff
@@ -0,0 +1,53 @@
+    <uses-permission android:name="android.permission.SCHEDULE_EXACT_ALARM"/>
+    <uses-permission android:name="android.permission.USE_EXACT_ALARM" />
+      <uses-permission android:name="android.permission.READ_MEDIA_IMAGES" />
+     <uses-permission android:name="android.permission.MANAGE_EXTERNAL_STORAGE" />
```
Fix indentation and spacing for better readability throughout the file.

Akhror commented on app/assets/data/locale.ts:
```diff
@@ -0,0 +1,132 @@
+        "locale": "uz"
+    },
+    {
+        "name": "Uzbekistan (Oʻzbekiston)",
```
`Uzbekistan (Узбекистан)` is more correct in Russian language.

Akhror commented on app/assets/translations/appOfflineData/getDataToStore.ts:
```diff
@@ -146,66 +146,136 @@ const getAllDataToStore = async (languageCode: string, dispatch: any, prevPage:
+        try {
+            await getDataToStore(languageCode, dispatch, BasicPagesSchema, Entity as BasicPagesEntity, basicPagesData, setAllTermsData);
+        } catch (error) {
+            console.log('error', error)
```
Use console.error instead of console.log for better debugging.
`console.error('Error fetching basic pages data:', error);`

Akhror commented on app/assets/translations/appOfflineData/getDataToStore.ts:
```diff
@@ -146,66 +146,136 @@ const getAllDataToStore = async (languageCode: string, dispatch: any, prevPage:
     let Entity: any;
-    if (apiEndpoint == appConfig.basicPages) {
-        await getDataToStore(languageCode, dispatch, BasicPagesSchema, Entity as BasicPagesEntity, basicPagesData, setAllTermsData);
+    if (apiEndpoint == appConfig.apiConfig.basicPages) {
```
Use strict equality (===)

Akhror commented on app/components/ChildDevelopmentCollapsibleItem.tsx:
```diff
@@ -1,4 +1,5 @@
-import { destinationFolder } from '@assets/translations/appOfflineData/apiConstants';
+// import { destinationFolder } from '@assets/translations/appOfflineData/apiConstants';
```
We can remove this line.

Akhror commented on app/components/FavActivities.tsx:
```diff
@@ -59,9 +57,7 @@ const FavActivities = (): any => {
-
-  const activityTaxonomyId = activeChild?.taxonomyData?.prematureTaxonomyId ?? activeChild?.taxonomyData?.id;
-
+  const activityTaxonomyId = activeChild?.taxonomyData.prematureTaxonomyId != null && activeChild?.taxonomyData.prematureTaxonomyId != undefined && activeChild?.taxonomyData.prematureTaxonomyId != "" ? activeChild?.taxonomyData.prematureTaxonomyId : activeChild?.taxonomyData.id;
```
Let's simplify:
```
const activityTaxonomyId = activeChild?.taxonomyData.prematureTaxonomyId 
    ? activeChild.taxonomyData.prematureTaxonomyId 
    : activeChild?.taxonomyData.id;
```

Akhror commented on app/components/HeaderBabyMenu.tsx:
```diff
@@ -149,40 +150,47 @@ const HeaderBabyMenu = (props: any): any => {
-                  {genderName != '' && genderName != null && genderName != undefined ? <Heading5 style={styles.heading5Fontwg}>{', ' + genderName}</Heading5> : 
-                  <Heading5 style={styles.heading5Fontwg}>{', ' + t('chilGender2')}</Heading5>
+                  {genderName != '' && genderName != null && genderName != undefined ? <Heading5 style={styles.heading5Fontwg}>{', ' + genderName?.name}</Heading5> :
+                    <Heading5 style={styles.heading5Fontwg}>{', ' + t('chilGender2')}</Heading5>
```
We can simplify:

```
<Heading5 style={styles.heading5Fontwg}>
    {', ' + (genderName ? genderName.name : t('chilGender2'))}
</Heading5>
```

Akhror commented on app/components/HeaderBabyMenu.tsx:
```diff
@@ -209,23 +217,23 @@ const HeaderBabyMenu = (props: any): any => {
               <ProfileSectionView>
                 <Heading3>{data.childName}
-                  {genderName != '' && genderName != null && genderName != undefined ? <Heading5 style={styles.heading5Fontwg}>{', ' + genderName}</Heading5> : null}
+                  {genderName != '' && genderName != null && genderName != undefined ? <Heading5 style={styles.heading5Fontwg}>{', ' + genderName?.name}</Heading5> : null}
```
Less code, same functionality:

`{genderName && <Heading5 style={styles.heading5Fontwg}>{', ' + genderName.name}</Heading5>}`

Akhror commented on app/components/HeaderBabyMenu.tsx:
```diff
@@ -341,11 +352,11 @@ const HeaderBabyMenu = (props: any): any => {
               source={{ uri: 'file://' + CHILDREN_PATH + activeChild.photoUri }}></ImageIcon>
           ) : (
-            activeChild.gender != '' ? activeChild.gender == '40' ? <Icon name="ic_baby" size={30} color={props.color || '#FFF'} /> : <Icon name="ic_baby_girl" size={30} color={props.color || '#FFF'} /> : <Icon name="ic_baby_girl" size={30} color={props.color || '#FFF'} />
+            activeChildGenderData != '' && activeChildGenderData?.unique_name != '' ? activeChildGenderData?.unique_name == taxonomyIds?.boyChildGender ? <Icon name="ic_baby" size={30} color={props.color || '#FFF'} /> : <Icon name="ic_baby_girl" size={30} color={props.color || '#FFF'} /> : <Icon name="ic_baby_girl" size={30} color={props.color || '#FFF'} />
```
Simplified version:
```

<Icon 
  name={
    activeChildGenderData?.unique_name && activeChildGenderData.unique_name === taxonomyIds?.boyChildGender
      ? "ic_baby"
      : "ic_baby_girl"
  } 
  size={30} 
  color={props.color || '#FFF'} 
/>

```

Akhror commented on app/components/growth/ChartWeightForHeight.tsx:
```diff
@@ -45,28 +50,28 @@ const ChartWeightForHeight = (props: any): any => {
   let obj: any;
   let standardDeviation: any;
-  if (activeChild?.gender == boyChildGender || activeChild?.gender == '') {
+  if (taxonomyIds?.boyChildGender == 'boy' || activeChild?.gender == '') {
```
Should we use appConfig.boyChildGender instead of row string?

Akhror commented on app/components/growth/ChartWeightForHeight.tsx:
```diff
@@ -117,7 +122,7 @@ const ChartWeightForHeight = (props: any): any => {
 
       <FlexCol>
-        {isChartVisible && deviceOrientation=='portrait' ? (
+        {isChartVisible && deviceOrientation == 'portrait' ? (
```
Ideally we should use constant values for all these hard coded string values.

Akhror commented on app/components/shared/RelatedActivities.tsx:
```diff
@@ -83,7 +84,7 @@ const RelatedActivities = (props:RelatedActivityProps):any => {
           actualselectedChildActivitiesData = randomArrayShuffle([...selectedChildActivitiesDatanew]);
         }
       // go not calclualte for growth screen
```
Let's remove or improve this comment.

Akhror commented on app/components/shared/RelatedVideoArticles.tsx:
```diff
@@ -61,7 +62,7 @@ const RelatedVideoArticles = (props: RelatedVideoArticlesProps): any => {
       state.utilsData.VideoArticlesData != '' ? JSON.parse(state.utilsData.VideoArticlesData) : [],
   );
-  const videoarticleDataold = VideoArticlesDataall.filter((x: any) => x.mandatory == videoArticleMandatory && x.child_age.includes(activeChild.taxonomyData.id) && (x.child_gender == activeChild?.gender || x.child_gender == bothChildGender));
+  const videoarticleDataold = VideoArticlesDataall.filter((x: any) => x.mandatory == appConfig.videoArticleMandatory && x.child_age.includes(activeChild.taxonomyData.id) && (x.child_gender == activeChild?.gender || x.child_gender == appConfig.bothChildGender));
```
Is VideoArticlesDataall existing code? It has spelling issues.

ansariDataamanulla replied:
yes it's existing code

Akhror commented on app/customHooks/useDigitConvert.ts:
```diff
@@ -0,0 +1,68 @@
+        'urd': ['۰', '۱', '۲', '۳', '۴', '۵', '۶', '۷', '۸', '۹'],
+        'ar': ['٠', '١', '٢', '٣', '٤', '٥', '٦', '٧', '٨', '٩'],
+        'tr': ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'],
+        'en': ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'], // Turkish uses Latin digits
```
This comment can be removed or placed next to 'tr' line.

Akhror commented on app/customHooks/useDigitConvert.ts:
```diff
@@ -0,0 +1,68 @@
+        // Add more languages and their digit mappings as needed
+    };
+
+    const [language, setLanguage] = useState<DigitLanguage>('bn'); // Default language is Bangla
```
Is default language Bengali, otherwise we need to correct or remove the comment.

Akhror commented on app/customHooks/useDigitConvert.ts:
```diff
@@ -0,0 +1,68 @@
+import { retry } from '@redux-saga/core/effects';
```
This import can be removed.

Akhror commented on app/instance/babuni/assets/offlinecontent.zip:
Remove archived folder.

Akhror commented on app/instance/babuni/assets/offlinecontent/ActivitiesData.ts:
```diff
@@ -0,0 +1,304 @@
+                491
+            ],
+            "cover_image": {
+                "url": "https://bangla.bebbo.app/sites/bangladesh/files/styles/content_1200xh_/public/2021-08/Who%2520Is%2520This%2520Person-cover-picture.jpg?itok=G9qcrQUl",
```
Let's change all https://bangla.bebbo.app to https://babuni.app. Eventually https://bangla.bebbo.app might get deleted.

Akhror replied:
Do you still need to resolve this comment?

Akhror replied:
Not resolved yet.

ansariDataamanulla replied:
resolved

Akhror commented on app/instance/babuni/assets/offlinecontent/ActivitiesData.ts:
```diff
@@ -0,0 +1,304 @@
+            "cover_image": {
+                "url": "https://bangla.bebbo.app/sites/bangladesh/files/styles/content_1200xh_/public/2021-08/flying_imagination-cover-picture.jpg?itok=QUjFCQE9",
+                "name": "flying_imagination-cover-picture.jpg",
+                "alt": "Illustration of Flying imagination"
```
Not sure how much attention we are giving to accessibility features but in good practice we should also translate alt text. throughout the file.

ansariDataamanulla replied:
@Akhror do it later Discussed with Evrim

Akhror commented on app/instance/babuni/assets/offlinecontent/countryData.ts:
```diff
@@ -0,0 +1,36 @@
+            "country_sponsor_logo": {
+              "url": "https://www.babuni.app/sites/bangladesh/files/styles/content_1200xh_/public/2024-05/MoH%20bangladesh.png?itok=LpBi5n0s",
+              "name": "MoH bangladesh.png",
+              "alt": "ministry of health Bangladesh"
```
Ministry of Health, Bangladesh

ansariDataamanulla replied:
@Akhror do it later Discussed with Evrim

Akhror commented on app/instance/bebbo/assets/images/chart_old.svg:
Is this file being used?

Akhror commented on app/instance/bebbo/assets/locale/BYru.ts:
```diff
@@ -0,0 +1,523 @@
+    loadingText: 'Подготовка контента…',
+    childSetupheader: 'Получите максимальную пользу от Беббо!',
+    childSetupdobLabel:
+      'Калі (у якую дату?) вы чакаеце дзіця (калі дзяця ужо нарадзілася, дадайце дату народзінаў)?',
```
This is not Russian language.

ansariDataamanulla replied:
done

Akhror commented on app/instance/bebbo/config/appConfig.ts:
```diff
@@ -0,0 +1,261 @@
+const config = {
+  destinationFolder: `${RNFS.DocumentDirectoryPath}/content/`,
+  buildForBebbo: 'bebbo',
+  buildFor: 'bebbo', // 'foleja'
```
Let's remove the comment.

ansariDataamanulla replied:
removed

Akhror commented on app/instance/bebbo/config/appConfig.ts:
```diff
@@ -0,0 +1,261 @@
+  videoTypeYoutube: 'youtube',
+  videoTypeImage: 'novideo',
+  backupGDriveFolderName: 'ParentBuddy',
+  backupGDriveFileName: 'mybackup.json',
```
This needs to be discussed with the team. I would recommend renaming it to "Bebbo-backup.json". 
We will update the other appConfig files accordingly if agreed.

ansariDataamanulla replied:
@Akhror do it later Discussed with Evrim

Akhror commented on app/instance/foleja/config/appConfig.ts:
```diff
@@ -0,0 +1,261 @@
+    }
+  },
+  allApisObject: (isDatetimeReq: any, dateTimeObj: any): any => {
+    const allApiObject = [
```
You can simplify by refactoring. Something like:

`const buildApiObject = (endpoint, saveinDB = true, isDatetimeReq = false, datetimeKey = '') => ({
  apiEndpoint: endpoint,
  method: 'get',
  postdata: isDatetimeReq && datetimeKey ? { datetime: datetimeKey } : {},
  saveinDB: saveinDB,
});`

Akhror commented on app/instance/merhabaBebek/assets/locale/tr.ts:
```diff
@@ -0,0 +1,458 @@
+    "developScreenrelatedArticleText": "İlgili tavsiye",
+    "developScreenrelatedAct": "İlgili oyunlar",
+    "developScreenviewDetails": "Devamını gör",
+    "articleScreenheaderTitle": "Tasiyeler",
```
This should be: "Tavsiyeler" instead of "Tasiyeler"

Akhror replied:
Not resolved yet

ansariDataamanulla replied:
resolved

Akhror commented on app/instance/bebbo/assets/locale/en.ts:
```diff
@@ -0,0 +1,531 @@
+    growthScreenenterDoctorRemarkTextPlaceHolder:
+      "Enter your doctor's remark or comment",
+    growthScreensaveMeasures: 'Save',
+    growthScreensaveMeasuresDetails: 'Save',
```
Maybe we can replcae growthScreensaveMeasuresDetails with growthScreensaveMeasures in the code.

Akhror commented on app/instance/merhabaBebek/assets/offlinecontent/ActivitiesData.ts:
```diff
@@ -0,0 +1,3375 @@
+            "cover_image": {
+                "url": "https://merhababebek.app/sites/turkey/files/styles/content_1200xh_/public/2021-08/the%2520box-cover-picture.jpg?itok=Ss0_XKgu",
+                "name": "the%20box-cover-picture.jpg",
+                "alt": "A carboard box with coloured papers inside"
```
Spelling: A cardboard

Akhror commented on app/instance/merhabaBebek/assets/standardDeviationOld.json:
Let's remove the file if not used.

Akhror commented on app/redux/reducers/localizationSlice.ts:
```diff
@@ -18,11 +18,11 @@ interface SelectedLocalizationType {
-  locale: localization[localization.length - 1]?.languages[0]?.locale, //'en'
-  pluralShow: localization[localization.length - 1]?.languages[0]?.pluralShow,//false
+  locale: 'en', //'en'
+  pluralShow: false,//false
```
Remove comments

Akhror commented on app/types/env.d.ts:
Should we delete this file?

Akhror commented on tsconfig.merhabaBebek.json:
Can we keep all tsconfig files for the instances under one folder or in their respective instance folders?

Akhror commented on app/instance/babuni/assets/images/activity_feature_logo.svg:
"Instance" folder should be "Instances".

Akhror commented on android/gradle.properties:
```diff
@@ -54,3 +54,12 @@ MYAPP_UPLOAD_STORE_FILE_FOLEJA=folejarelease.keystore
+MYAPP_UPLOAD_STORE_FILE_MERHABA_BEBEK=merhabaBebek.keystore
+MYAPP_UPLOAD_KEY_ALIAS_MERHABA_BEBEK=merhabaBebekalias
+MYAPP_UPLOAD_STORE_PASSWORD__MERHABA_BEBEK=$pass$123
+MYAPP_UPLOAD_KEY_PASSWORD__MERHABA_BEBEK=$pass$123
```
Consider enabling the new architecture and R8 for better performance.


```
#  Improves React Native’s performance and modularity.
newArchEnabled=true 

# R8 optimizes, shrinks, and obfuscates code, reducing APK size.
android.enableR8=true 
```

ansariDataamanulla replied:
@Akhror we ll do it later

Akhror commented on android/gradle.properties:
```diff
@@ -54,3 +54,12 @@ MYAPP_UPLOAD_STORE_FILE_FOLEJA=folejarelease.keystore
+MYAPP_UPLOAD_STORE_FILE_MERHABA_BEBEK=merhabaBebek.keystore
+MYAPP_UPLOAD_KEY_ALIAS_MERHABA_BEBEK=merhabaBebekalias
+MYAPP_UPLOAD_STORE_PASSWORD__MERHABA_BEBEK=$pass$123
+MYAPP_UPLOAD_KEY_PASSWORD__MERHABA_BEBEK=$pass$123
```
Use environment variables (local.properties or .env) to avoid exposing credentials in version control.

`MYAPP_UPLOAD_STORE_PASSWORD=${SYSTEM_ENV_STORE_PASSWORD}`

ansariDataamanulla replied:
@Akhror we ll do it later

Akhror commented on app/types/env.d.ts:
```diff
@@ -1,8 +1,13 @@
+  export const nameProject: string;
+  export const projectNumber: string;
+  export const webId: string;
+  export const iosId: string;
```
Let's make sure to add webId and iosId description into readme.md by relating that they are webClientId and iosClientId.

```
webClientId - is the OAuth 2.0 Client ID for web applications. It is required for authenticating users on Android using Firebase or Google Sign-In. It allows silent authentication and offline access.

iosClientId -  is the OAuth 2.0 Client ID specifically for iOS apps. Required for signing in with Google on iOS using Firebase or Google Sign-In. 
```

Akhror commented on ios/InfoPlist_Bebbo/bn-BD.lproj/InfoPlist.strings:
```diff
@@ -0,0 +1,8 @@
+  Created by Kalyani Kaushik on 28/01/22.
+  
+*/
+"CFBundleDisplayName" = "Babuni";
```
Correct me. My understanding is this file should not be in Bebbo folder.

Akhror replied:
I do not think if this was resolved.

ansariDataamanulla replied:
resolved

Akhror commented on ios/InfoPlist_MerhabaBebek/tr.lproj/InfoPlist.strings:
```diff
@@ -0,0 +1,8 @@
+  InfoPlist.strings
+  ParentBuddyApp
+
+  Created by Kalyani Kaushik on 28/01/22.
```
Was this note added automatically?

Akhror commented on ios/ParentBuddyApp.xcodeproj/xcuserdata/ansari.xcuserdatad/xcschemes/ParentBuddyAppBangla copy.xcscheme:
These user data files should be excluded from the repository.

Akhror commented on ios/ParentBuddyApp.xcodeproj/xcuserdata/ansari.xcuserdatad/xcschemes/xcschememanagement.plist:
These user data files should be excluded from the repository.

Akhror commented on ios/ParentBuddyApp/Info.plist:
```diff
@@ -33,7 +37,7 @@
 			<key>CFBundleURLSchemes</key>
 			<array>
-				<string>com.googleusercontent.apps.$(projectNumber)-$(clientIdKey)</string>
+				<string>com.googleusercontent.apps.185520625819-2paqi14b84eaov6d69t2d2tnno5fri3p</string>
```
I believe this file should be excluded from the project. Otherwise we should get clientIdKey from .env files.

ansariDataamanulla replied:
I will check do it later

Akhror replied:
Please check with Evrim, we cannot post this key to public repo.

Akhror commented on ios/MerhabaBebek-Info.plist:
```diff
@@ -0,0 +1,123 @@
+	</dict>
+	<key>NSCameraUsageDescription</key>
+	<string>This app requires camera permission for capturing profile image</string>
+	<key>NSLocationWhenInUseUsageDescription</key>
```
We can remove this key since Bebbo does not use location services.

Akhror commented on ios/MerhabaBebek-Info.plist:
```diff
@@ -0,0 +1,123 @@
+			<string>Bundle ID</string>
+			<key>CFBundleURLSchemes</key>
+			<array>
+				<string>com.googleusercontent.apps.30528917983-r05tab92r1ulpl0hmipn19kug00864ve</string>
```
This  Google OAuth client ID key should not be exposed to public

Akhror commented on app/services/Utils.ts:
```diff
@@ -25,20 +25,21 @@ import { receiveAPIFailure } from "../redux/sagaMiddleware/sagaSlice";
+import moment from "moment";
+const requestNotificationPermission = async (): Promise<any> => {
+    const status = await requestNotifications([]);
+    console.log(status, "..status..");
```
Let's please make sure this kind of logs do not go to production.

Akhror commented on app/services/backup.tsx:
```diff
@@ -70,77 +71,91 @@ class Backup {
+                            isBase64: false,
+                        });
+                        if (typeof response !== 'string') {
+                            return false;
```
If createFileMultipart() fails and response is not a string, function returns false without handling it properly.  You may need try-catch/

```
.then(async (cipher: any) => {
    try {
        const response = await googleDrive.createFileMultipart({
            name: appConfig.backupGDriveFileName,
            content: cipher.cipher,
            parentFolderId: backupFolderId,
            isBase64: false,
        });

        return typeof response === 'string';
    } catch (error) {
        console.error('Error uploading to Google Drive:', error);
        return false;
    }
})
.catch((error: any) => {
    console.error('Error exporting data:', error);
    return false; 
});
```

Akhror commented on app/services/backup.tsx:
```diff
@@ -70,77 +71,91 @@ class Backup {
-    public async importFromFile(oldChildrenData: any, navigation: any, genders: any, dispatch: any, childAge: any, langCode: any): Promise<any> {
-        console.log("oldchildrenresponse nwq",oldChildrenData)
+    public async importFromFile(oldChildrenData: any, navigation: any, genders: any, dispatch: any, childAge: any, langCode: any, taxonomyIds?: any): Promise<any> {
+        console.log("oldchildrenresponse nwq", oldChildrenData)
```
Let's exclude in prod

ansariDataamanulla replied:
we already set up configuration all console exclude in prod

Akhror commented on app/services/backup.tsx:
```diff
@@ -151,38 +166,24 @@ class Backup {
-            })
+                }).catch(error => {
+                    console.log("error-", error);
+                    return new Error('No Import Succeded');
```
1. It should be "Succeeded" not "Succeded".
2. Instead of returning error you should throw.

```
.catch(error => {
    console.error("Error occurred:", error);
    throw new Error('No Import Succeeded');
});
```

Akhror commented on app/services/commonApiService.ts:
```diff
@@ -258,11 +266,17 @@ export const onHomeapiSuccess = async (response: any, dispatch: any, navigation:
-        dispatch(setDownloadedBufferAgeBracket(artarray))
+        try {
+          const storedata = store.getState();
+          const childagearray = storedata.utilsData.taxonomy.allTaxonomyData != '' ? JSON.parse(storedata.utilsData.taxonomy.allTaxonomyData).child_age : [];
```
If allTaxonomyData is null, undefined, or an empty object, the condition might not work as expected.
JSON.parse() without try-catch can crash the app.

```
let childagearray = [];
try {
    if (storedata.utilsData.taxonomy.allTaxonomyData) {
        childagearray = JSON.parse(storedata.utilsData.taxonomy.allTaxonomyData).child_age || [];
    }
} catch (error) {
    console.error("JSON Parsing Error:", error);
    childagearray = []; 
} 
```

Akhror commented on app/services/googleAuth.ts:
```diff
@@ -23,65 +23,69 @@ class GoogleAuth {
         } catch (error) {
-            console.log("error-",error)
-         }
+            console.log("error-qqqq", error)
```
Improve or remove the log.

Akhror commented on app/services/googleDrive.ts:
```diff
@@ -4,8 +4,8 @@ import {
-import { backupGDriveFolderName } from "@assets/translations/appOfflineData/apiConstants";
-
+import { appConfig } from "../instance";
+import { projectNumber, webId, iosId } from 'react-native-dotenv';
```
Remove unused imports.

Akhror commented on app/services/healthCheckupService.ts:
```diff
@@ -14,31 +17,32 @@ export const getAllHealthCheckupPeriods = ():any => {
-    // getall doctor's place measures
-  const allMeasures = activeChild?.measures.filter((item:any)=>(item.measurementPlace==0)).sort(
+  // filter by measurementPlace  is Doctors and sort by measurementDate
+  // getall doctor's place measures
```
Let's improve the comment as much as possible. Otherwise we can remove.

Akhror commented on app/services/healthCheckupService.ts:
```diff
@@ -51,58 +55,69 @@ const checkIfMeasuredVaccineExistsForLocale = (vaccineIds:any):any=>{
-    if (hcItem) {
-      const measure = allMeasurements.filter((measure:any) => (measure.childAgeInDaysForMeasure >= hcItem?.vaccination_opens) && (measure.childAgeInDaysForMeasure < hcItem?.vaccination_ends))
+    if (hcItem != '') {
+      const measure = allMeasurements.filter((measure: any) => (measure.childAgeInDaysForMeasure >= hcItem?.vaccination_opens) && (measure.childAgeInDaysForMeasure < hcItem?.vaccination_ends))
```
```
if (hcItem && hcItem.vaccination_opens !== undefined && hcItem.vaccination_ends !== undefined) { 
    const measure = allMeasurements.filter(
        (measure: any) =>
            (measure?.childAgeInDaysForMeasure ?? 0) >= hcItem.vaccination_opens &&
            (measure?.childAgeInDaysForMeasure ?? 0) < hcItem.vaccination_ends
    );
}

```

Akhror commented on app/services/periodicSync.ts:
```diff
@@ -118,23 +118,23 @@ export const getAllPeriodicSyncData = (): any => {
-              ageBrackets.push(ages);
-            })
+    if (childList.length > 0) {
+      childList.map((child: any) => {
```
Since you're not returning a new array, .forEach() is the better choice over .map().

Akhror commented on ios/ParentBuddyApp.xcworkspace/xcuserdata/ansari.xcuserdatad/WorkspaceSettings.xcsettings:
Should we exclude entire xcuserdata folder from repo?

Akhror commented on ios/ParentBuddyApp.xcodeproj/xcuserdata/ansari.xcuserdatad/xcschemes/MerhabaBebek.xcscheme:
Should we exclude entire xcuserdata folder from repo?

Akhror replied:
Not resolved yet.

ansariDataamanulla commented:
Code Restructuring:
The file and folder structure has been updated to follow an instance-based structure, making it easier to manage and build instances. This update improves organization and scalability, allowing for faster, more efficient builds on an instance-wise basis.
Updated and changed the scope of Google Drive backup integration to enhance storage efficiency and backup reliability.
Unique Name Changes:
Merged the Babuni implementation with the Bebbo codebase, making necessary adjustments to align with the new naming convention for taxonomyIds.
Bug Fixes:
Fixed font rendering issues across different devices for consistent typography.
Resolved minor bugs affecting the overall user experience, improving stability and performance.
